build_projects/dotnet-cli-build/MakeRelative.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.IO;
+using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+using Microsoft.DotNet.Cli.Build.Framework;
+using Newtonsoft.Json;
+
+namespace Microsoft.DotNet.Cli.Build
+{
+    public class MakeRelative : Task
+    {
+        [Required]
+        public string Path1 { get; set; }
+
+        [Required]
+        public string Path2 { get; set; }
+
+        public char SeperatorChar { get; set; }
+
+        [Output]
+        public ITaskItem RelativePath { get; set; }
```

Comment:
Maintaining the input context makes it easier to do Shared-Metadata batching in some scenarios e.g. when you want to correlate a relative path to a full path.